### PR TITLE
Add List Bucket Policies API

### LINKED ISSUE #611: Implement list bucket policies
This issue has been created to track and finish work defined in https://github.com/minio/minio-py/pull/508

### PR DESCRIPTION
**Pull request by vadmeste**
Fixes #611

### CODE REVIEW

**Comment by vadmeste** [comment]
This is not WIP anymore

**Comment by poornas** [comment]
@vadmeste, the bucket policy listing seems to be incorrect - can you please check?

```
➜  minio-py git:(bbf4670) ✗ mc policy list myminio/u22
u22/p3/images/* => readwrite
u22/p2/test/* => readonly
u22/p3/test/* => writeonly
➜  minio-py git:(bbf4670) ✗ python list_bucket_policies.py
{u'u22/p3/test/*': 'none', u'u22/p3/images/*': 'none', u'u22/p2/test/*': 'readonly'}
{u'u22/p3/test/*': 'none', u'u22/p3/images/*': 'none', u'u22/p2/test/*': 'readonly'}
➜  minio-py git:(bbf4670) ✗
```

**Comment by deekoder** [comment]
@poornas  can you give a X so that it shows up as requested changes.

File: minio/api.py [hunk 1]
```diff
@@ -437,6 +437,30 @@ def set_bucket_policy(self, bucket_name, prefix, policy_access):
                            body=content,
                            content_sha256=content_sha256_hex)
```
Comment (poornas) [inline minio/api.py]: function comment is missing.

**Comment by deekoder** [comment]
ping @vadmeste  any progress on this PR so that reviewers can re-review.

**Comment by vadmeste** [comment]
@deekoder this was blocked because of another bug https://github.com/minio/minio-py/issues/630 but since we are simplifying the API, we will remove this API. So this PR is no more needed. I already checked with @krishnasrinivas.

**Pull request closed without merging**